### ISSUE TITLE
`phpcbf --standard=psr2` strips `?>` in the middle of files

### ISSUE THREAD

**Issue by aaronjameslang** [squizlabs/PHP_CodeSniffer#824]
`phpcbf --standard=psr2` strips `?>` in the middle of files, which breaks things

This:
`for(...) { ?> <some-html> <?php }`
becomes this
`for(...) { <some-html> <?php }`
and fails to parse

PHP_CodeSniffer version 2.3.3 (stable) by Squiz

**Comment by aaronjameslang** [comment]
But only some times

```
@@ -181,13 +181,12 @@ class Dog
         <?php
-            if (!$readOnly) {
-        ?>
-                            <div><div/>
-        <?php
-            } else {
-                $colspan=5;
-            }
+        if (!$readOnly) {
+                        <div><div/>
+            <?php
+        } else {
+            $colspan=5;
+        }
         ?>

**Comment by Konafets** [comment]
I would consider this as correct. 

From PSR-2: 

    Code MUST follow a "coding style guide" PSR-1.

From PSR-1:

    Files MUST use only <?php and <?= tags.

**Comment by aaronjameslang** [comment]
It should be shown as an error by phpcs, but phpcbf should only fix the things it is capable of fixing. By removing '?>' it breaks things, meaning that I can't use it on a codebase without a lot of manual vetting.

**Comment by gsherwood** [comment]
This was fixed in version 2.4.0. Please upgrade to the latest version (2.5.0) to get this and other fixes.

In 2.5.0, this code:
```php
for($i=0;$i<10;$i++) { ?> <some-html> <?php }
```
Gets fixed to this:
```php
for ($i=0; $i<10; $i++) {
?> <some-html> <?php
}
```
